frontend/app/components/SurveyComponent.jsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+import React, { useState } from "react";
+import { Model } from "survey-core";
+import { Survey } from "survey-react-ui";
+import "survey-core/defaultV2.min.css";
+import theme from "../../public/utils/survey_theme.json";
+
+function SurveyComponent({ data, assessmentID, patientID }) {
+    const [survey] = useState(new Model(data));
+    const [pageNo, setPageNo] = useState(survey.currentPageNo);
+    const [isRunning, setIsRunning] = useState(true);
+
+    survey.applyTheme(theme);
+
+    survey.onCurrentPageChanged.add((_, options) => {
+        setPageNo(options.newCurrentPage.visibleIndex);
+    });
+
+    survey.onStarted.add(() => { setIsRunning(true); } );
+
+    survey.onComplete.add(() => { setIsRunning(false); });
+
+    survey.onComplete.add((sender, options) => {
+        options.showSaveInProgress();
+
+        let surveryResults = sender.data;
+
+        survey.getAllQuestions().map((question) => {
+            if(!surveryResults[question.name]){
+                surveryResults[question.name] = "";
+            }
+        });
+
+        const formattedResults = formatSurveyResults(surveryResults);
+
+        // send formatted survey results
+        // options.showSaveError();
+        const currentDate = new Date();
+        const dataToSend = {
+            assessment_id: assessmentID,
+            patient_id: patientID,
+            data: formattedResults,
+            created_on: currentDate.toISOString().slice(0, -1)//  + (currentDate.getMilliseconds() / 1000).toFixed(6).slice(1),
+        }
+
+        console.log(JSON.stringify(dataToSend))
```

Comment:
in the future we can remove it once we have API.

Reply:
Yes this was just to show I was getting the responses